File: video capture testing/cvtest/cvtest.cpp
#include <iostream>
#include <opencv2/opencv.hpp>
#include <chrono>

using namespace std;
using namespace std::chrono;
using namespace cv;

typedef high_resolution_clock Clock;


int main()
{
    VideoCapture cap;
    Mat image;
    bool ret;
    Clock::time_point t, t0;
    duration<double> totalElapsed;
    double totalElapseds;
    int max_counter = 100;
    int counter = 0;

    cap.setExceptionMode(true);

    vector<int> params = { VideoCaptureProperties::CAP_PROP_FRAME_WIDTH, 1920, VideoCaptureProperties::CAP_PROP_FRAME_HEIGHT, 1080 };
    try {
        if (!cap.open(0, VideoCaptureAPIs::CAP_ANY, params)) {
            cout << "Unable to open camera" << endl;
            return -1;
        }

        cout << "Camera back end: " << cap.getBackendName() << endl;
        if (!cap.isOpened()) {
            cout << "Unable to open camera" << endl;
            return -1;
        }
        //cap.set(VideoCaptureProperties::CAP_PROP_FRAME_WIDTH, 1920);
        //cap.set(VideoCaptureProperties::CAP_PROP_FRAME_HEIGHT, 1080);

        t0 = high_resolution_clock::now();

        while (true) {
            ret = cap.read(image);
            //cap >> image;
            if (counter == max_counter) {
                t = high_resolution_clock::now();
                totalElapsed = (t - t0) / counter;
                totalElapseds = totalElapsed.count();
                t0 = t;
                counter = 0;
                cout << totalElapseds << endl;
            }
            counter++;
        }
    } catch (cv::Exception& e) {
        cout << "Exception " << e.what() << endl;
    }
}

File: video capture testing/python/cvtest.py
import cv2 as cv
import time


if __name__ == '__main__':
    params = [cv.CAP_PROP_FRAME_WIDTH, 1920, cv.CAP_PROP_FRAME_HEIGHT, 1080]
    cap = cv.VideoCapture()
    if not cap.open(0, cv.CAP_ANY, params):
        raise IOError("Unable to open camera")
    print("Camera back end:", cap.getBackendName())
    if not cap.isOpened():
        raise IOError("Unable to open camera")
    #cap.set(cv.CAP_PROP_FRAME_WIDTH, 1920)
    #cap.set(cv.CAP_PROP_FRAME_HEIGHT, 1080)

    max_counter = 100
    counter = 0
    t0 = time.process_time()

    while True:
        ret, image = cap.read()
        if counter == max_counter:
            t = time.process_time()
            elapsed_time = (t - t0) / counter
            t0 = t
            counter = 0
            print(elapsed_time)
        counter += 1

    cap.release()
